### PR TITLE
[8] Feat/interactiveobject

### PR DESCRIPTION
**Pull request by Tanguylo**
Refactor of draw methods
Refactor of drawn objects
Refactor of mousable objects

### CODE REVIEW

**Comment by Tanguylo** [comment]
First step of refactoring TS in view of performance and UX works

File: src/shapes.ts [hunk 1]
```diff
@@ -298,41 +337,57 @@ export class Text extends Shape {
...
   }
 
+  private splitInWords(): string[] {
+    if (this.text.length == 0) return [""];
+    return this.buildWords();
+  }
+
+  private isTextTooWide(context: CanvasRenderingContext2D, text: string): boolean {
+    return context.measureText(text).width > this.boundingBox.size.x
+  }
```
Comment (Tanguylo) [inline src/shapes.ts:371]: change to isTooWide

Comment (GhislainJ) [reply src/shapes.ts:371]: Idea : Consider `textOverflows` ?

Comment (Tanguylo) [reply src/shapes.ts:371]: Yours is better :)

**Pull request merged**